.github/workflows/e2e-test-train-api.yaml

```diff
@@ -0,0 +1,54 @@
+name: E2E Test with train API
+on:
+  - pull_request
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes-version: ["v1.28.7"]
+        python-version: ["3.9", "3.10", "3.11"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Free-Up Disk Space
+        uses: ./.github/workflows/free-up-disk-space
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.10.0
+        with:
+          node_image: kindest/node:${{ matrix.kubernetes-version }}
+          cluster_name: training-operator-cluster
+          kubectl_version: ${{ matrix.kubernetes-version }}
+
+      - name: Build training-operator
+        run: |
+          ./scripts/gha/build-image.sh
+        env:
+          TRAINING_CI_IMAGE: kubeflowtraining/training-operator:test
+
+      - name: Deploy training operator
+        run: |
+          ./scripts/gha/setup-training-operator.sh
+        env:
+          KIND_CLUSTER: training-operator-cluster
+          TRAINING_CI_IMAGE: kubeflowtraining/training-operator:test
+          GANG_SCHEDULER_NAME: "none"
+          KUBERNETES_VERSION: ${{ matrix.kubernetes-version }}
+
+      - name: Run tests
+        run: |
+          pip install pytest
+          python3 -m pip install -e sdk/python -r sdk/python/kubeflow/trainer/requirements.txt; pytest -s sdk/python/test_train_api/test_e2e_train_api.py --log-cli-level=debug
```

Comment:
@helenxie-bit Please can you move your e2e test into:
`/sdk/python/test/e2e-train-api/test_e2e_train_api.py`
And modify the pytest directory here to sdk/python/test/e2e
https://github.com/kubeflow/training-operator/blob/master/.github/workflows/integration-tests.yaml#L99
In that case, it will be easier for us to find all different E2Es that we run.

Reply:
@andreyvelich I have moved the file as suggested, please have a look when you have time.